### DISCUSSION TITLE
Actix Web 4 - default_service SPA

### DISCUSSION THREAD

**Discussion by StefanRupertsberger** [actix/actix-web#2473] [Q&A] [answered]
I want to serve a SPA with actix-web and I used to do it like this: [Example](https://github.com/rishadbaniya/Actix-web-SPA-react-js-example).
But this does not work anymore with actix-web 4.
The thing that does not work is to redirect every unknown path to the `index.html` of the SPA to use angular routing.

**Comment by staninprague** [comment, marked as answer]
Here is a possible solution, not tested, only proof of concept that compiles:

```
use actix::fut::ok;
use actix_web::dev::ServiceRequest;
....

 .default_service(|r: ServiceRequest| {
      ok(crate::react_router::spa_index(r))
}),
```

And spa_index function:

```
pub (crate) fn spa_index(service_request: ServiceRequest) -> ServiceResponse
{
    let (req, _payload) = service_request.into_parts();

    let file_response = NamedFile::open("./static/index.html");

    ServiceResponse::new(req.clone(), file_response.respond_to(&req))
}
```

Hope it might help. I have much more elaborated spa_index, so only crafted the above as a theoretical exercise that compiles and looks like it might be working.
